### PR TITLE
Add single Node API

### PR DESCRIPTION
**Pull request by tal-az**
only api, to be integrated with UI

### CODE REVIEW

File: controllers/agents.js [hunk 1]
```diff
@@ -7,10 +7,15 @@ async function addNode(req, res) {
         //validate hostId
         const hostId = req.params.hostId
         const osType = req.params.osType;
+        const username = req.body.username;
+        const password = req.body.password;
+        const privateKey = req.body.privateKey;
         if (!hostId || !osType) {
-            return res.status(400).send({ error: 'hostID or osType parameter is required' });
+            return res.status(400).send({ error: 'hostID and osType parameter is required' });
```
Comment (GeneralSnuggles) [inline controllers/agents.js]: are*

Comment (GeneralSnuggles) [reply controllers/agents.js]: Shouldn't we provide an error per parameter?
Why give such a general one?

File: controllers/agents.js [hunk 2]
```diff
@@ -7,10 +7,15 @@ async function addNode(req, res) {
...
+        }
+        if (!password && !privateKey) {
+            return res.status(400).send({ error: 'either password or privateKey parameter is required' });
```
Comment (GeneralSnuggles) [inline controllers/agents.js:20]: Same here, should just show the problem

File: services/system.js [hunk 3]
```diff
@@ -1,40 +1,91 @@
...
+const fleetctlBashDownload = fs.readFileSync('./scripts/install_fleet.sh').toString()
 
-const fleetctlBashDownload = fs.readFileSync('./scripts/install_fleet.sh')
 const secret_path = process.env.SSH_SECRET_PATH
 const username = process.env.SSH_USERNAME
+
+async function getSSHCon(host, username, password = null, privateKey = null) {
+  const ssh = new NodeSSH();
+  if (!password && !privateKey) {
+    throw new Error('Either "password" or "privateKey" parameter is required.');
```
Comment (GeneralSnuggles) [inline services/system.js:15]: Same as before

File: services/system.js [hunk 4]
```diff
@@ -1,40 +1,91 @@
...
+  try{
+    console.log("[*] Enrolling Fleet agent on " + host_id)
+    let response = await executeRemoteCommand(sshConn, fleetctlBashDownload);
+    response = await executeRemoteCommand(sshConn, './fleetctl -v')
+    response = await executeRemoteCommand(sshConn, "./" + enrollCmd)
+    if (password){
+      response = await executeRemoteCommand(sshConn, path.join("echo ", password, " | sudo -S dpkg -i fleet-osquery*.deb"))
+    }
+    else{
+      response = await executeRemoteCommand(sshConn, "sudo -S dpkg -i fleet-osquery*.deb")
```
Comment (GeneralSnuggles) [inline services/system.js]: Running sudo without a password?

File: services/system.js [hunk 5]
```diff
@@ -1,40 +1,91 @@
...
+    // exec('ssh -o StrictHostKeyChecking=no -i ' + secret_path + ' ' + username + '@' + host_id  + ' \"./' + enrollCmd + ' sudo dpkg -i fleet-osquery*.deb\"', (error, stdout, stderr) => {
+    // // exec('ssh -o StrictHostKeyChecking=no -i ' + secret_path + ' ' + username + '@' + host_id  + ' \"sudo dpkg -i fleet-osquery*.deb\"', (error, stdout, stderr) => { //If you already downloaded the pkg and want it to go faster, tests only
+    //     if (error) {
+    //       console.error(`exec error: ${error}`);
+    //       return;
+    //     }
+    //     // console.log(`stdout: ${stdout}`);
+    //     // console.error(`stderr: ${stderr}`);
+    //   })
+    // // validate response as successful or failed
```
Comment (GeneralSnuggles) [inline services/system.js]: What's all of these comments

File: controllers/agents.js (hunk 2 above)
Comment (tal-az) [reply controllers/agents.js:20]: at least one of them is required. this error shows if none is submitted so it's okay.

File: services/system.js (hunk 3 above)
Comment (tal-az) [reply services/system.js:15]: same as before

File: services/system.js (hunk 4 above)
Comment (tal-az) [reply services/system.js:51]: whoops, missed the echo from the first if. thanks

Comment (tal-az) [reply services/system.js]: feature, if password is not submitted modification of soduers for installation is required. will be documented. https://biosquery.atlassian.net/browse/BQU-118

**Review by GeneralSnuggles — APPROVED** [review]
Nice

**Pull request merged**